MST/CMakeLists.txt

```diff
@@ -0,0 +1,215 @@
+################
+# Iman Jehanzeb
+################
+
+cmake_minimum_required(VERSION 3.18)
+
+cmake_host_system_information(RESULT Ncpu QUERY NUMBER_OF_PHYSICAL_CORES)
+message(STATUS "CMake ${CMAKE_VERSION} with max. ${Ncpu} threads")
+
+project(MST LANGUAGES Fortran C CXX CUDA)
+
+# Defining mst_main
+add_executable(mst_main)
+
+set_target_properties(mst_main PROPERTIES OUTPUT_NAME "MST")
+
+set_target_properties(mst_main PROPERTIES
+        RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin"
+        )
+
+
+# Defining mst_lib 
+add_library(mst_lib)
+
+set_target_properties(mst_lib PROPERTIES
+        LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib
+        ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib
+        Fortran_MODULE_DIRECTORY ${CMAKE_BINARY_DIR}/modules)
+        
+target_include_directories(mst_lib PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+
+target_link_libraries(mst_main PUBLIC mst_lib)
+target_link_libraries(mst_main PUBLIC gfortran)
+
+
+# Adding Subdirectories
+add_subdirectory(src)
+add_subdirectory(lib)
+add_subdirectory(plib)
+add_subdirectory(iolib)
+
+
+
+# Find MPI Library
+find_package(MPI REQUIRED COMPONENTS CXX OPTIONAL_COMPONENTS)
```

Comment:
Normally, you only need Fortran. 

`find_package(MPI REQUIRED COMPONENTS Fortran OPTIONAL_COMPONENTS)`